### PR TITLE
Internal NO_BACKSLASH_ESCAPES support

### PR DESCRIPTION
**Pull request by lauxjpn**
I included the test (console) project I used to test the functionality before implementing this PR against MySqlConnector and Connector/ODBC.
These need to be integrated into the actual test suite. ~~I just can't get this to run without VS 16.3 (which will be released with .NET Core 3.0 on Monday, i presume).~~
Generally, all existing tests should work with `NO_BACKSLASH_ESCAPES` enabled as well.

This addresses #701 and https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/827

### CODE REVIEW

**Comment by lauxjpn** [comment]
I got the tests running on my machine now. I will therefor add a couple of test cases to the suite and remove the console test project from this PR.

File: src/MySqlConnector/Core/SqlParser.cs [hunk 1]
```diff
@@ -265,6 +270,10 @@ public void Parse(string sql)
 				states |= FinalParseStates.NeedsNewline;
 				state = beforeCommentState;
 			}
+			else if (state == State.SingleQuotedStringSingleQuote)
+			{
+				state = State.Statement;
+			}
```
Comment (bgrainger) [inline src/MySqlConnector/Core/SqlParser.cs]: This is a general class of error I can push a fix for (before this PR is merged).

**Comment by bgrainger** [comment]
The Travis CI tests time out for PRs. I haven't figured out why this is happening (they just seem to run uniformly slower, and I don't know why), but you can ignore them.

Long-term, I plan to migrate to Azure Pipelines. Short-term, I suppose I could just disable Travis CI on PRs, since I know it'll fail.

**Comment by lauxjpn** [comment]
>The Travis CI tests time out for PRs. I haven't figured out why this is happening (they just seem to run uniformly slower, and I don't know why), but you can ignore them.

I'll take a quick look at it, if I can figure out why this is happening there.

File: src/MySqlConnector/Core/SqlParser.cs (hunk 1 above)
Comment (bgrainger) [reply src/MySqlConnector/Core/SqlParser.cs]: 5060d9d5c5267cb7480b299bed4bc948a3dcad03; will merge to `master` soon.

**Comment by lauxjpn** [comment]
The test suite just seems to take to long. Nothing really hangs, the individual tests are just getting executed extremely slow. I have very little experience with Travis CI, so I can't say if this is expected.

The tests could be split into multiple test ENVs and then executed separately/parallel.

But it's probably best to just make the switch to Azure DevOps, as @caleblloyd just did here: https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/pull/823

**Comment by lauxjpn** [comment]
Okay, this came up in a search, regarding Travis CI:

https://github.com/travis-ci/travis-ci/issues/3049#issuecomment-215084577
https://github.com/travis-ci/travis-cookbooks/issues/888

**Comment by bgrainger** [comment]
Those are interesting links, but it seems like that problem has been ongoing for two (or more) years. From PR history, I know that Travis was working in April, but was timing out in May, so the change (for this project) happened in that timeframe. 

I need to review what Travis is testing that I haven't replicated on Azure Pipelines and either finish copying it, or decide to leave it untested.

**Comment by lauxjpn** [comment]
I ended up disabling most of the tests on Travis CI.
Do the added test scenarios suffice, or should I add some in a particular area?

**Comment by bgrainger** [comment]
The new tests look good--thanks! Do you have any preference on how I merge/squash/rebase this?

**Comment by lauxjpn** [comment]
Just squash it.

**Comment by bgrainger** [comment]
Merged via d550a69688628cd5a56fb29e38ff0dcb415267f9.

**Pull request closed without merging**